tortoise/queryset.py

```diff
@@ -453,59 +453,65 @@ def order_by(self, *orderings: str) -> "QuerySet[MODEL]":
         queryset._orderings = new_ordering
         return queryset
 
-    def latest(self, *orderings: str) -> "QuerySetSingle[Optional[MODEL]]":
+    def _single_queryset(
```

Comment:
Does it really make sense to have this method? It does not reduce the number of lines really but makes the logic more complicated because it has logical branches.

It replaces pretty clean code which is more like configuration than logic:

<img width="478" alt="Screenshot 2024-11-11 at 00 35 08" src="https://github.com/user-attachments/assets/cafd7a7e-e3ab-4783-a7ca-f60e2d1fe33f">

Reply:
@henadzit You can see the difference here: https://github.com/waketzheng/tortoise-orm/commit/d340109a51cc0a03ec0d8c0dad0bd1dbc31ea78b
There are 5 lines reduced, and 4 `# type: ignore` marks removed.

Reply:
It reduces the code base by 5 lines but adds quite a lot of complexity to the code. It either clones a queryset or modifies it in place. Imagine, if the Python's standard `sorted` function would sometimes return a new sorted list and in some cases sort the list in place. It would be so weird! I feels to me not worth it.

Reply:
@henadzit I agree that simple is better than complex. I refactor the code to use a new function `_as_single`. https://github.com/tortoise/tortoise-orm/pull/1758/commits/ec0ea01fab6a0b06d15bd2e580db3478fa747fe6

Reply:
Thank you for collaborating with me on this!